File: eBarberShop/eBarberShop.Model/Novosti.cs
namespace eBarberShop.Model
{
    public class Novosti
    {
        public int NovostiId { get; set; }
        public string Naslov { get; set; }
        public string Sadrzaj { get; set; }
        public DateTime DatumObjave { get; set; }
        public byte[]? Slika { get; set; }
        public int KorisnikId { get; set; }
        public Korisnici Korisnik { get; set; }

        public string? KorisnikImePrezime => Korisnik?.Ime + " " + Korisnik?.Prezime;
    }
}

File: eBarberShop/UI/ebarbershop_desktop/lib/models/novosti/novosti.dart
import 'package:json_annotation/json_annotation.dart';

part 'novosti.g.dart';

@JsonSerializable()
class Novosti {
  int novostiId;
  String naslov;
  String sadrzaj;
  DateTime datumObjave;
  String? slika;
  int korisnikId;
  String? korisnikImePrezime;

  Novosti(this.novostiId, this.naslov, this.sadrzaj, this.datumObjave, this.slika, this.korisnikId, this.korisnikImePrezime);

  factory Novosti.fromJson(Map<String, dynamic> json) =>
      _$NovostiFromJson(json);

  Map<String, dynamic> toJson() => _$NovostiToJson(this);
}
